jq.pyx

```diff
@@ -51,6 +60,55 @@ cdef extern from "jq.h":
     void jq_get_error_cb(jq_state *, jq_err_cb *, void **)
 
 
+cdef object _jv_to_python(jv v):
+    """Unpack a jv value into a Python object"""
+    cdef jv_kind kind = jv_get_kind(v)
+    cdef int i
+    cdef jv ik
+    cdef jv iv
+    if kind == JV_KIND_INVALID:
+        raise ValueError("Invalid value")
+    elif kind == JV_KIND_NULL:
+        return None
+    elif kind == JV_KIND_FALSE:
+        return False
+    elif kind == JV_KIND_TRUE:
+        return True
+    elif kind == JV_KIND_NUMBER:
+        if jv_is_integer(v):
+            return int(jv_number_value(v))
+        else:
+            return float(jv_number_value(v))
+    elif kind == JV_KIND_STRING:
+        return jv_string_value(v).decode("utf-8")
+    elif kind == JV_KIND_ARRAY:
+        arr = []
+        for i in range(0, jv_array_length(jv_copy(v))):
+            iv = jv_array_get(jv_copy(v), i)
+            try:
+                arr.append(_jv_to_python(iv))
+            finally:
+                jv_free(iv)
```

Comment:
Would changing the contract of `_jv_to_python` to consume its input simplify some of this code?

Reply:
Yep. I considered this before in context of memory usage, but couldn't wrap my head around it. Did some experiments and amended it. Should be better now.

Also, I love your usage of "contract" :grinning: Nobody would understand me if I tried talking like that at my place :disappointed:.